From 1ef27d22f786babf093a96e570c7f46d3d1a0b36 Mon Sep 17 00:00:00 2001
From: Ingo Molnar <mingo@kernel.org>
Date: Mon, 9 Oct 2023 17:31:26 +0200
Subject: [PATCH] sched/topology: Move the declaration of 'schedutil_gov' to
 kernel/sched/sched.h

References: bsc#1212887 (Scheduler functional and performance backports)
Git-commit: f2273f4e19e29f7d0be6a2393f18369cd1b496c8
Patch-mainline: v6.7-rc1

Move it out of the .c file into the shared scheduler-internal header file,
to gain type-checking.

Signed-off-by: Ingo Molnar <mingo@kernel.org>
Cc: Shrikanth Hegde <sshegde@linux.vnet.ibm.com>
Cc: Valentin Schneider <vschneid@redhat.com>
Link: https://lore.kernel.org/r/20231009060037.170765-3-sshegde@linux.vnet.ibm.com
Signed-off-by: Mel Gorman <mgorman@suse.de>
---
 kernel/sched/sched.h    | 2 ++
 kernel/sched/topology.c | 1 -
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/kernel/sched/sched.h b/kernel/sched/sched.h
index 8f539eae2bb8..788fa6e5cb91 100644
--- a/kernel/sched/sched.h
+++ b/kernel/sched/sched.h
@@ -3160,6 +3160,8 @@ static inline bool sched_energy_enabled(void)
 	return static_branch_unlikely(&sched_energy_present);
 }
 
+extern struct cpufreq_governor schedutil_gov;
+
 #else /* ! (CONFIG_ENERGY_MODEL && CONFIG_CPU_FREQ_GOV_SCHEDUTIL) */
 
 #define perf_domain_span(pd) NULL
diff --git a/kernel/sched/topology.c b/kernel/sched/topology.c
index 9d14fafa55e3..5d1a044f146c 100644
--- a/kernel/sched/topology.c
+++ b/kernel/sched/topology.c
@@ -212,7 +212,6 @@ static unsigned int sysctl_sched_energy_aware = 1;
 static DEFINE_MUTEX(sched_energy_mutex);
 static bool sched_energy_update;
 
-extern struct cpufreq_governor schedutil_gov;
 static bool sched_is_eas_possible(const struct cpumask *cpu_mask)
 {
 	bool any_asym_capacity = false;
